### ISSUE TITLE
[Bug]: Subscription to WhenStatusChanged() not properly disposed on Disconnect

### ISSUE THREAD

**Issue by Glomby** [shinyorg/shiny#801]
### Component

BluetoothLE Client

### What platform(s) are effected?

- [X] iOS 12
- [X] iOS 13
- [X] iOS 14
- [X] Android 8
- [X] Android 9
- [X] Android 10
- [X] Android 11
- [ ] UWP (Sponsors ONLY)

### Steps To Reproduce

1. Scan for Advertisements of a specific device.
2. Connect to the device and subscribe to the WhenStatusChanged(), status should be "connected".
3. Disconnect the device.
4. Scan again and repeat multiple times. For each time the WhenStatusChanged() is called multiple times. I.E. On the secand time "Disconnected" will be triggered twice and after the third connection it will be trigger three times.

### Expected Behavior

WhenStatusChanged() should only report one Connection Event at a time.

### Actual Behavior

WhenStatusChanged() reports once on the first connection, twice on the second and so on.

### Exception or Log output

No Exceptions.

I print the Connect Status whenever it changes and for each connection I get more and more of these events. 

### Code Sample

To start scanning I call:

```
if (!bleManager.IsScanning && watcher == null)
{
    watcher = bleManager
        .Scan()
        .Buffer(TimeSpan.FromSeconds(1))
        .Where(x => x?.Any() ?? false)
        .Subscribe(results => { Advertisement_Received(results); }, exc => Debug.WriteLine(exc.ToString()));
}
``` 

In the "Advertisement_Received" I connect to the device like this:
```
bleManager.StopScan();
this.watcher?.DisposeWith(Disposable);
                                
peripheral.Connect(new ConnectionConfig() { AutoConnect = false, AndroidConnectionPriority = ConnectionPriority.High });

disposable_conn_failed = peripheral.WhenConnectionFailed().Subscribe(ble_exc => conn_failed(ble_exc));
disposable_conn_updated = peripheral.WhenStatusChanged().Subscribe(status => status_changed(status));
```

When the connection status changes to disconnected I dispose of eall the characteristics I'm using and the scanner (watcher):

```
if (watcher != null)
       watcher?.DisposeWith(Disposable);

if (peripheral != null)
{
    peripheral.CancelConnection();
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aritchie** [comment]
You aren't disposing of something properly then.  It won't report multiple times otherwise. 

You'll need to provide a FULL example as attachment if you feel otherwise

**Comment by Glomby** [comment]
Ok I'll create a reduced example project for you.

**Comment by Glomby** [comment]
Okay I've created an example project with only the necessary functionality but for simplicity sake I changed all the "DisposeWith()" calls to "Dispose()" and the error didn't appear.

I've now done the same in my actual project and again the problem disappeared. So it was definitly a problem with objects not being disposed properly but due to using DisposeWith and not using Dispose().

So this problem has been resolved.

Is there a specific reason why this was caused this way? I followed the original examples of the old Xamarin.BLE Plugin (not shiny) when implementing all the DisposeWith() calls and only recently ported everything to Shiny.BLE. And I'm sure this is how you Disposed most of the Objects back then.
